Code/GraphMol/CIPLabeler/CIPLabeler.cpp

```diff
@@ -0,0 +1,193 @@
+        atom_cfg = Tetrahedral::RIGHT;
+      } else {
+        atom_cfg = Tetrahedral::LEFT;
+      }
+
+      std::unique_ptr<Tetrahedral> cfg{
+          new Tetrahedral(mol, atom, std::move(carriers), atom_cfg)};
+      configs.push_back(std::move(cfg));
+    }
+  }
+
+  for (auto i = 0u; i < mol.getNumBonds(); ++i) {
+    auto bond = mol.getBond(i);
+
+    int bond_cfg = -1;
+    switch (bond->getStereo()) {
+    case Bond::STEREOE:
+    case Bond::STEREOTRANS:
+      bond_cfg = Sp2Bond::OPPOSITE;
+      break;
+    case Bond::STEREOZ:
+    case Bond::STEREOCIS:
+      bond_cfg = Sp2Bond::TOGETHER;
+      break;
+    default:
+      continue;
+    }
+
+    auto stereo_atoms = MolOps::findStereoAtoms(bond);
+    std::vector<Atom *> anchors{
+        {mol.getAtom(stereo_atoms[0]), mol.getAtom(stereo_atoms[1])}};
+    std::vector<Atom *> atoms{{bond->getBeginAtom(), bond->getEndAtom()}};
+
+    std::unique_ptr<Sp2Bond> cfg(
+        new Sp2Bond(mol, bond, std::move(atoms), std::move(anchors), bond_cfg));
+    configs.push_back(std::move(cfg));
+  }
+
+  return configs;
+}
+
+bool labelAux(const std::vector<std::unique_ptr<Configuration>> &configs,
+              const Rules &rules,
+              const std::unique_ptr<Configuration> &center) {
+  using Node_Cfg_Pair = std::pair<Node *, Configuration *>;
+  std::vector<Node_Cfg_Pair> aux;
+
+  auto &digraph = center->getDigraph();
+  for (const auto &config : configs) {
+    if (config == center) {
+      continue;
+    }
+    // FIXME: specific to each descriptor
+    const auto &foci = config->getFoci();
+    for (const auto &node : digraph.getNodes(foci[0])) {
+      if (node->isDuplicate()) {
+        continue;
+      }
+      auto low = node;
+      if (foci.size() == 2) {
+        for (const auto &edge : node->getEdges(foci[1])) {
+          const auto &other_node = edge->getOther(node);
+          if (other_node->getDistance() < node->getDistance())
+            low = other_node;
+        }
+      }
+      if (!low->isDuplicate()) {
+        aux.emplace_back(low, config.get());
+      }
+    }
+  }
+
+  auto pair_cmp = [](const Node_Cfg_Pair &a, const Node_Cfg_Pair &b) {
+    return a.first->getDistance() > b.first->getDistance();
+  };
+  std::sort(aux.begin(), aux.end(), pair_cmp);
+
+  boost::unordered_map<Node *, Descriptor> queue;
+  int prev = std::numeric_limits<int>::max();
+  for (const auto &e : aux) {
+    const auto &node = e.first;
+
+    if (node->getDistance() < prev) {
+      for (const auto &e2 : queue) {
+        e2.first->setAux(e2.second);
+      }
+      queue.clear();
+      prev = node->getDistance();
+    }
+    const auto &config = e.second;
+    auto label = config->label(node, digraph, rules);
+    queue.emplace(node, label);
+  }
+
+  for (const auto &e : queue) {
+    e.first->setAux(e.second);
+  }
+
+  return true;
+}
+
+void label(CIPMol &mol,
+           const std::vector<std::unique_ptr<Configuration>> &configs) {
```

Comment:
again, you're changing the things pointed to by the elements of `configs`, so I think it shouldn't be `const`

Reply:
Also same as previous comment: this was true until the `Digraph` got owned by the configuration -- nothing else changes on the configs (labels are set on the mol, which is not `const`).